homeassistant/components/shelly/config_flow.py

```diff
@@ -262,6 +264,53 @@ async def async_step_confirm_discovery(
             errors=errors,
         )
 
+    async def async_step_reauth(self, entry_data: Mapping[str, Any]) -> FlowResult:
+        """Handle configuration by re-auth."""
+        self.entry = self.hass.config_entries.async_get_entry(self.context["entry_id"])
+        return await self.async_step_reauth_confirm()
+
+    async def async_step_reauth_confirm(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Dialog that informs the user that reauth is required."""
+        errors: dict[str, str] = {}
+        assert self.entry is not None
+        host = self.entry.data[CONF_HOST]
+
+        if user_input is not None:
+            info = await self._async_get_info(host)
```

Comment:
Don't we need to catch errors here?

Reply:
Yes, we need to catch errors here, similar to
https://github.com/bieniu/home-assistant-core/blob/863de282871dad3ee35abe6793f99394e64db589/homeassistant/components/shelly/config_flow.py#L207-L212

Reply:
Oh you're right. I'll open a PR to fix this.

Reply:
Done https://github.com/home-assistant/core/pull/79205